middleware/reportflow.go

```diff
@@ -0,0 +1,35 @@
+package middleware
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/goyek/goyek/v2"
+)
+
+// ReportStatus is a middleware which reports the flow execution status.
+//
+// The format is based on the reports provided by the Go test runner.
+func ReportFlow(next goyek.Executor) goyek.Executor {
+	return func(in goyek.ExecuteInput) error {
+		out := in.Output
+
+		from := time.Now()
+		err := next(in)
+		if _, ok := err.(*goyek.FailError); ok {
+			fmt.Fprintf(out, "%v\t%.3fs\n", err, time.Since(from).Seconds())
+			return err
+		}
+		if err == context.Canceled || err == context.DeadlineExceeded {
```

Comment:
```suggestion
		if errors.Is(err, context.Canceled) || errors.Is(err, context.DeadlineExceeded) {
```
Something could wrap the error at some point and this will be KO

Reply:
I also think that this check makes only sense to make this check in `Flow.main` to assign a "proper" status code.
https://github.com/goyek/goyek/pull/420/commits/1f6eabed4815780cb9b34a2eae87104bb1939745